ptvsd/messaging.py

```diff
@@ -34,6 +34,11 @@ def from_stdio(cls):
                 msvcrt.setmode(stdout.fileno(), os.O_BINARY)
         return cls(stdin, stdout)
 
+    @classmethod
+    def from_socket(cls, socket):
+        socket_io = socket.makefile('rwb', 0)
```

Comment:
>makefile [](start = 27, length = 8)

Should we check if the socket has timeout set? From 2.7.15 docs for socket.makefile(): "The socket must be in blocking mode (it can not have a timeout)."

Reply:
Yep, and it does in fact break spectacularly if you try! It's easy to check, too.